src/plugins/dashboard/public/application/top_nav/dashboard_top_nav.tsx

```diff
@@ -604,6 +604,8 @@ export function DashboardTopNav({
                   data-test-subj="dashboardAddPanelButton"
                 />
               ),
+              addControlButton:
+                dashboardAppState.dashboardContainer.controlGroup?.getControlButton('toolbar'),
```

Comment:
Note that, if the new controls are not enabled in Labs, then `controlGroup` won't be defined - therefore, this toolbar button *won't* be rendered! No extra checks necessary - cool! 👍